Support tags on volume snapshots

NiR- opened the pull request.

koalalorenzo commented:
Hello! Thanks for your contribution! 
It looks like the new functions are lacking tests! Could you add it?

koalalorenzo commented on digitalocean/Tag.py:
```diff
@@ -79,29 +80,33 @@ def __remove_resources(self, resources):
         resources = []
         if not isinstance(data, list): return data
-        for a_droplet in data:
+        for obj in data:
```
was it really necessary to rename to a generic `obj` instead of `a_droplet`?

NiR- replied:
> was it really necessary to rename to a generic obj instead of a_droplet?

Again, this method is now more generic than before: it can be used to build `resources` field for any resources that support it, so it does not make sense to keep `a_droplet`.

> Please if you have to rename, think about something that another maintainer can understand without the need of an IDE or a deep knowledge of the source code/api.

This is a variable inside a method, how does that influence the API?

I agree this name is not perfect, but I can't find something that's both as generic as the method is and not too generic at the same time. If you come with something better, tell me.

koalalorenzo replied:
> This is a variable inside a method, how does that influence the API?
One of the principle that I am trying to push in this library is to make it understandable even for a junior. So `obj` is no good :( What if we rename it into `api_generic_response`? Will this new name help a new developer? Does it make sense to you?  😅 

Thank you btw for your help 👍 😄

NiR- replied:
> One of the principle that I am trying to push in this library is to make it understandable even for a junior on its first day of work. So obj is no good :(

Got it, sounds like a reasonable expectation actually :smile: 

> What if we rename it into api_generic_response? Will this new name help a new developer? Does it make sense?

Well, I'm not sure it's a better name :| 1. We're not manipulating responses in this method, but DTOs implemented by this lib ; 2. The name is a bit cumbersome IMHO.
To be honest, I'm not totally happy with that `resources` variable too (the one built by the method).

I came up with `obj` because `resources` was already used, but this would have been the best fit IMHO. But then we need to differentiate the input resources and the resources API field we build. What we can do to enhance both names would be to rename the current `resources` into `resources_field` and `obj` into `resource_to_tag`. WDYT?

koalalorenzo replied:
sounds like a great idea! 👍

koalalorenzo commented on digitalocean/Tag.py:
```diff
@@ -79,29 +80,33 @@ def __remove_resources(self, resources):
 
 
-    def __extract_resources_from_droplets(self, data):
+    def __build_resources(self, data, object_class: str, resource_type: str):
```
Was it really necessary to rename the method?

NiR- replied:
> Was it really necessary to rename the method?

Since this method is now reused to build other `resources` fields, yeah it makes sense to rename this method to something more generic, that don't include "droplet" in its name.

koalalorenzo replied:
Got it! :) We can make it even more clear in the docs then!

NiR- commented:
Yeah sorry, forgot to add tests as I'm working on a dependent project that don't have any :man_facepalming: I'm adding tests right now :wink:

NiR- commented:
I updated the API mocks for tag tests as they were not on par with the API anymore. I also pushed the requested tests, but in a fixup commit, so I will need to rebase and force-push my branch once you approve it.

I updated the doc block for `__build_resources_field` but I'm waiting on your comment(s) to rename the `obj` var and push these fixes together :wink:

NiR- commented:
I pushed the last fix this morning. Tests are passing and everything seems to run as expected in the depending project :smile:

koalalorenzo commented:
LGTM! I will merge it now :) Thank you so much for helping!

NiR- commented:
Is there any chance you could release a new tag? This would prevent the depending software I'm working on to require a specific git commit.
